internal/ingest/ingest.go

```diff
@@ -156,7 +157,7 @@ func NewIngester(cfg config.Ingest, h host.Host, idxr indexer.Interface, reg *re
 
 		providersBeingProcessed: make(map[peer.ID]chan struct{}),
 		providerAdChainStaging:  make(map[peer.ID]*atomic.Value),
-		toWorkers:               make(chan providerID),
+		toWorkers:               make(chan providerID, 1000),
```

Comment:
What does this solve by putting the providers in the channel without a worker available?  Having the channel unbuffered caused backpressure on ingestion if there are no workers available.  With this change, I am concerned that it ends up loading 1000 providers into the channel, and then causing back-pressure as before.  That would use up more memory without solving any problem.